### PR TITLE
Removes model objects

### PR DESCRIPTION
**Pull request by janakerman**
#37 

Horrible change. No typing. Feels... wrong. Why?
Whyyyy?

Tests are currently broken because the model objects
previously had equals logic to simply compare IDs. Now
we're returning the raw JSON we'll need to do that equality
checking somewhere else.

### CODE REVIEW

**Comment by janakerman** [comment]
Pushed this up so that change is visible and can be discussed. @bjedrzejewski 

I don't like this at all - so little typing - too dynamic for my liking. Can't we all just Java?

**Comment by janakerman** [comment]
also fixes #39

File: src/main/java/com/chessmates/controller/EntityController.groovy [hunk 1]
```diff
@@ -22,12 +20,12 @@ class EntityController {
      * Return a list of all team players
      */
     @GetMapping(value = 'players')
-    List<Player> getPlayers() { entityService.getPlayers() }
+    List getPlayers() { entityService.getPlayers() }
```
Comment (bjedrzejewski) [inline src/main/java/com/chessmates/controller/EntityController.groovy:23]: Why did you decide to drop the Generic here completely? Can't we have something more detailed than Object? We at least know that this is Json I think?

Comment (bjedrzejewski) [reply src/main/java/com/chessmates/controller/EntityController.groovy:23]: Ok, I see, thi sis not the way in Groovy. Perhaps maybe we should  be clear that this is actually getPlayersJson() method?

File: src/main/java/com/chessmates/controller/EntityController.groovy [hunk 2]
```diff
@@ -22,12 +20,12 @@ class EntityController {
...
 
     /**
      * Return a list of all unique games played between all team members.
      */
     @GetMapping(value = 'games')
-    List<Game> getGames() { entityService.getGames() }
+    List getGames() { entityService.getGames() }
```
Comment (bjedrzejewski) [inline src/main/java/com/chessmates/controller/EntityController.groovy:29]: Same here maybe- getGamesJson- what do you think?

File: src/main/java/com/chessmates/lichess/data/LichessApi.groovy [hunk 3]
```diff
@@ -11,15 +7,15 @@ interface LichessApi {
...
      * Get the players of a given Lichess team.
      */
-    LichessResultPage<Player> getPlayers(String teamId, int pageNumber, int pageSize)
+    def getPlayers(String teamId, int pageNumber, int pageSize)
 
     /**
      * Get games of a given player.
      */
-    LichessResultPage<Game> getGames(String playerId, int pageNumber, int pageSize)
+    def getGames(String playerId, int pageNumber, int pageSize)
```
Comment (bjedrzejewski) [inline src/main/java/com/chessmates/lichess/data/LichessApi.groovy]: LichessResultPage as a wrapper is not a bad idea, this is a wrapper that contains multiple GamesJson I think. We could just abstract that GamesJson as an Object, but shape of the ResultPage could be defined?

File: src/main/java/com/chessmates/lichess/data/LichessDataServiceImpl.groovy [hunk 4]
```diff
@@ -47,20 +44,25 @@ class LichessDataServiceImpl implements LichessDataService {
...
+                fetchPlayerPage.curry(TEAM_NAME),
                 stopAtPlayerId
         )
 
-        final players = resultSet.get()
-
-        players.each { playerRepository.save it }
+        List players = resultSet.get()
+                .stream()
+                .filter(LichessEntityValidator.&isValid)
```
Comment (bjedrzejewski) [inline src/main/java/com/chessmates/lichess/data/LichessDataServiceImpl.groovy:62]: Nice 👍

File: src/main/java/com/chessmates/repository/MetaDataRepository.groovy [hunk 5]
```diff
@@ -1,27 +1,24 @@
...
 
     /** Save the latest game requested from the Lichess API for a given set of opponents. */
-    void saveLatestGame(Player player, Player opponent, Game game)
+    void saveLatestGame(player, opponent, game)
 
     /** Get the latest game request from the Lichess API for a given set of opponents.
      * ImmutableMap is returned to indicate that changes to this store don't affect the store contents.
      */
-    ImmutableMap<ImmutablePair<Player, Player>, Game> getLatestGames()
+    ImmutableMap getLatestGames()
```
Comment (bjedrzejewski) [inline src/main/java/com/chessmates/repository/MetaDataRepository.groovy]: I don't think we had to lose ImmutablePair to Object mapping here. I think keeping it is a good idea in the signature.

File: src/main/java/com/chessmates/repository/MockGameRepository.groovy [hunk 6]
```diff
@@ -13,22 +11,22 @@ import org.springframework.stereotype.Repository
...
     @Override
-    void save(Game game) {
+    void save(Object game) {
         logger.debug "(Mock) Saving game: ${game}"
         store.put(game.id, game)
     }
 
     @Override
-    Game find(String gameId) {
+    Object find(String gameId) {
```
Comment (bjedrzejewski) [inline src/main/java/com/chessmates/repository/MockGameRepository.groovy]: We are not consistent here. Sometimes we use Object, sometimes we leave it empty. It is a super minor comment.

**Review by bjedrzejewski — APPROVED** [review]
Good job, I have a few comments and suggestions that you might think about to improve the general readability here.

File: src/main/java/com/chessmates/lichess/data/LichessApi.groovy (hunk 3 above)
Comment (janakerman) [reply src/main/java/com/chessmates/lichess/data/LichessApi.groovy]: Agreed - this needs to be renamed to make it clear that we're returning a page of data here, to differentiate it from the getGames methods that return a list of games.

File: src/main/java/com/chessmates/controller/EntityController.groovy (hunk 2 above)
Comment (janakerman) [reply src/main/java/com/chessmates/controller/EntityController.groovy:29]: As discussed it's not really JSON - I'll change the LichessApi to be more explicit about returning a page.

**Pull request merged**